DEVELOPERS.md

```diff
@@ -0,0 +1,203 @@
+# Development on puppet-agent module
+
+Welcome to the `puppet-agent` module! This is a document which will help
+guide you to getting started on development. The key sections here are:
+
+1. [Iterative code changes](#iterative)
+1. [Important files to understand repo](#files)
+1. [Tools for development](#tools)
+1. [Testing](#testing)
+
+# Iterative code changes<a name="iterative"/>
+
+/*Scott*/
+For iterative development on Ubuntu using Docker, see:
+- docker/ubuntu/Dockerfile
+- docker/bin/run-ubuntu.sh
+
+/*Casey*/
+1. Check out a vmpooler VM.
+2. Manually install PE on it (download from getpe.delivery.puppetlabs.net)
+3. Manually install local copy of the puppet_agent module on it
+4. Request another VM (the agent)
+5. Use the puppet_agent bolt task to install an agent on second VM
+6. Associate the master with an agent
+7. Write a manifest to test what I'm working on
+8. Run `puppet apply` with that manifest
+9. Edit the module on the master, and repeat.
+
+/*Branan*/
+1. Forward your ssh key to do github operations
+2. Ensure editor suppports editing remote files
+3. Checkout a vmpooler machine
+4. Clone the module
+5. Run `puppet apply` to test it as I work
+
+/*Enis*/
```

Comment:
@ekinanp Any differences you'd want to see here?

Reply:
From discussion today, development for FOSS only is fine for the majority of cases. I have added a stub for the case where a developer is working on a platform that is PE-specific.

For the example manifests, are you picturing something that could live in a doc/ folder? If the manifests are super short and relevant for popular development cases, putting them in here sounds fine.

Reply:
Did you push the changes? I don't see anything that's been updated.

The manifests should be pretty short, so I think placing them here would be fine.

Reply:
I put it in a section below:
https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/357/files#diff-710af9c85139525b79a31decd9f92c91R63